### PR TITLE
Draft intro about accessibility, user agents, and the curb cut effect

### PR DESCRIPTION
**Pull request by pavpanchekha**
This PR adds an introductory section to the accessibility chapter that tries to tie accessibility to the concepts of user agents, interoperability, and the principles of the web, and introduces environmental and temporary disabilities. I tried making it "inspiring" instead of "preachy", but let me know if that succeeded.

### CODE REVIEW

File: book/accessibility.md [hunk 1]
```diff
@@ -5,23 +5,80 @@ prev: animations
...
 
 [declare]: intro.md#browser-code-concepts
 
 [a11y]: https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility
 
+What is accessibility
+=====================
+
+"Accessibility" refers to any way a user can change or customize how
+they interact with a web page,[^not-just-screen-reader] and the web's
```
Comment (chrishtr) [inline book/accessibility.md:26]:
```suggestion
they interact with a web page in order to make it to use,[^not-just-screen-reader] and the web's
```

Comment (chrishtr) [reply book/accessibility.md:26]: Add another footnote acknowledging other definitions, such as "the practice of making your websites usable by as many people as possible" [1] (which is a framing of the same thing from a web developer point of view

[1] https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility

File: book/accessibility.md [hunk 2]
```diff
@@ -5,23 +5,80 @@ prev: animations
...
+layout, and rendering of a web page, as well as interact with a web
+page with their keyboard, by voice, or using some kind of helper
+software.
+
+The reasons for customizing, of course, are as diverse as the
+customizations themselves. For example, when my son was born,[^pavel]
+my wife and I alternated time taking care of the baby and I ended up
+spending a lot of time working at night. To maximize precious sleep,
+I wanted the screen to be less bright, and was thankful that many
+websites offer a "dark mode". Later, I found that taking notes by
```
[the remark this answers is not in the document]
Comment (chrishtr) [reply book/accessibility.md:38]: I think it's better to omit the (not intended as, I know) scare quotes here, like you omitted for screen readers below.

File: book/accessibility.md [hunk 3]
```diff
@@ -5,23 +5,99 @@ prev: animations
...
+[declarative] UI and the flexibility of HTML and CSS to make it
+possible to interact with the web page by touch, keyboard, or voice.
+
+[ua]: intro.md#the-role-of-the-browser
 
 [declare]: intro.md#browser-code-concepts
 
 [a11y]: https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility
 
+What is accessibility
```
Comment (chrishtr) [inline book/accessibility.md:22]:
```suggestion
What is accessibility?
```

File: book/accessibility.md [hunk 4]
```diff
@@ -5,23 +5,99 @@ prev: animations
...
+[^toddler-curb-cut]: And when my son grows a bit and starts walking on
+    his own, he'll likely still be small enough that walking up a curb
+    without a curb cut will be difficult for him.
+    
+[^for-now]: For now, that is---perhaps software assistants will become
+more widespread as software improves, mediating between the user and
+web pages. Password managers and form autofill agents are already
+somewhat like this.
+
+Of course, accessibility *does* include advanced accessibility tools
```
Comment (chrishtr) [inline book/accessibility.md:79]:
```suggestion
Of course, accessibility *does* include advanced tools
```

File: book/accessibility.md [hunk 5]
```diff
@@ -5,23 +5,99 @@ prev: animations
...
+like screen readers, because these tools bring enormous benefits to
```
Comment (chrishtr) [inline book/accessibility.md:80]: Link to a definition of screen reader here?

File: book/accessibility.md [hunk 6]
```diff
@@ -5,23 +5,99 @@ prev: animations
...
+those that need them.[^moral] But the common lesson of all kinds of
+accessibility work, physical and digital, is that once an
+accessibility tool is built, creative people find that it helps in all
+kinds of situations unforseen by the tool's designers. Dark mode helps
+you tell your work and personal email apart; web page zoom helps you
+print the whole web page on a single sheet of paper; and keyboard
+shortcuts let you leverage muscle memory to submit many similar orders
+to a web application that doesn't have a batch mode.
+
+[^moral]: And we have a moral responsibility to help, plus the
```
Comment (chrishtr) [inline book/accessibility.md:90]:
```suggestion
[^moral]: And we have an ethical responsibility to help all users.
```

File: book/accessibility.md [hunk 7]
```diff
@@ -5,23 +5,99 @@ prev: animations
...
+    commercial incentive of serving as large a market or reaching as
```
Comment (chrishtr) [inline book/accessibility.md:91]:
```suggestion
```

File: book/accessibility.md [hunk 8]
```diff
@@ -5,23 +5,99 @@ prev: animations
...
+    large an audience as possible.
```
Comment (chrishtr) [inline book/accessibility.md:92]:
```suggestion
```

Comment (chrishtr) [reply book/accessibility.md:92]: (I don't like mixing ethics with capitalism in the same sentence.)

File: book/accessibility.md [hunk 9]
```diff
@@ -5,23 +5,99 @@ prev: animations
...
+
+Moreover, accessibility derives from the same [principles](intro.md)
+that birthed the web: user control, multi-modal content, and
+interoperability. These principles allowed the web to be accessible to
+different browsers and different operating systems, and these same
```
Comment (chrishtr) [inline book/accessibility.md:97]: This sentence says "different" four times - can you rephrase?

File: book/accessibility.md (hunk 8 above)
Comment (pavpanchekha) [reply book/accessibility.md:92]: How about making it less "capitalistic" but keeping the core point, something like this:

> We have an ethical responsibility to help all users. Plus, there is the practical matter that if you're making a web page, it's presumably because you want as many people as possible using it.

File: book/accessibility.md (hunk 9 above)
Comment (pavpanchekha) [reply book/accessibility.md:97]: Hmm, I was hoping to emphasize a parallelism between what you might call "machine accessibility" and "human accessibility". Does that not come across?

Comment (chrishtr) [reply book/accessibility.md:100]: How about:

"These principles allowed the web to be accessible to
different browsers and operating systems---and *these same principles* likewise make the web accessible to people of all types and abilities.

File: book/accessibility.md (hunk 8 above)
Comment (chrishtr) [reply book/accessibility.md]: How about rephrasing the second clause to "you want as many people as possible to benefit from it."

Comment (chrishtr) [reply book/accessibility.md]: (without "presumably")

Comment (pavpanchekha) [reply book/accessibility.md]: Great!

File: book/accessibility.md (hunk 9 above)
Comment (pavpanchekha) [reply book/accessibility.md]: Done

File: book/accessibility.md [hunk 10]
```diff
@@ -5,23 +5,102 @@ prev: animations
...
+interact with a web page in order to make it easier to
+use.[^other-defs][^not-just-screen-reader] The web's uniquely-flexible
+core technologies mean that browsers offer a lot of accessibility
+features[^not-just-screen-reader] to allow a user to customize the
+style, layout, and rendering of a web page, as well as interact with a
+web page with their keyboard, by voice, or using some kind of helper
+software.
+
+[^other-defs]: Accessibility can also be defined from the developer's
+    point of view, [in which case][mdn-def] it's something like the
```
Comment (chrishtr) [inline book/accessibility.md:35]:
```suggestion
    point of view, [in which case][mdn-def] it's something like: the
```

File: book/accessibility.md [hunk 11]
```diff
@@ -5,23 +5,102 @@ prev: animations
...
+accessibility tool is built, creative people find that it helps in all
+kinds of situations unforseen by the tool's designers. Dark mode helps
+you tell your work and personal email apart; web page zoom helps you
+print the whole web page on a single sheet of paper; and keyboard
+shortcuts let you leverage muscle memory to submit many similar orders
+to a web application that doesn't have a batch mode.
+
+[^moral]: We have an ethical responsibility to help all users. Plus,
+    there is the practical matter that if you're making a web page,
+    it's because you want as many people as possible to benefit from
```
Comment (chrishtr) [inline book/accessibility.md:94]:
```suggestion
    you want as many people as possible to benefit from
```

**Review by chrishtr — APPROVED** [review]
